src/realm/sync/noinst/client_history_impl.hpp

```diff
@@ -381,8 +381,7 @@ class ClientHistory final : public _impl::History, public TransformHistory {
     // ServerHistory object.
     mutable util::Optional<Arrays> m_arrays;
 
-    mutable std::vector<char> m_changeset_from_server_owner;
-    mutable util::Optional<HistoryEntry> m_changeset_from_server;
+    mutable const HistoryEntry* m_changeset_from_server = nullptr;
```

Comment:
Since this variable is a pointer to a local variable that gets reset during `commit()` - i.e. the thing this points to has a weird lifetime - can you add a comment explaining what this is and when it gets set and reset?

Reply:
I refactored things so that the history entry is added in integrate_server_changesets() and the member variable is just a bool (and also added some comments).